x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -357,16 +359,15 @@ export function getWaterfall(
   const duration = getWaterfallDuration(items);
   const serviceColors = getServiceColors(items);
 
-  const entryTransaction = entryWaterfallTransaction?.doc;
-
   return {
-    entryTransaction,
+    entryWaterfallTransaction,
     rootTransaction,
     duration,
     items,
     errorsPerTransaction,
     errorsCount: errorItems.length,
     serviceColors,
     errorItems,
+    childrenByParentId: getChildrenGroupedByParentId(items),
```

Comment:
@sqren WDYT about this? I didn't want to change the whole logic to order the trace items, so I kept it as it is today, `getOrderedWaterfallItems`, and just grouped the sorted items.

Reply:
I don't see any problems with it. Do you have any concerns in particular about this?

Reply:
no, I only wanted your feedback, I really rather not changing the `getOrderedWaterfallItems`.